### PR TITLE
Add react app

### PR DESCRIPTION
**Pull request by dudewheresmycode**
Laying the groundwork for the react layer. I was able to get the whole electron app + react dev app, to start with `npm start` from the project root. Which is super handy!

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "react-redo",
```
Comment (vanstinator) [inline client/package.json]: this should probably be `client` so the workspace matches

File: tsconfig.json [hunk 2]
```diff
@@ -1,12 +1,16 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
+    "module": "esnext",
```
Comment (vanstinator) [inline tsconfig.json]: this _might_ bite us with electron. but I hope not.

File: package.json [hunk 3]
```diff
@@ -4,23 +4,31 @@
...
+    "typescript": "^4.4.4",
+    "wait-on": "^6.0.0"
   },
   "dependencies": {
     "deepspeech": "^0.9.3",
     "express": "^4.17.1",
     "ffmpeg-static": "^4.4.0"
+  },
+  "volta": {
+    "node": "16.13.0"
```
Comment (vanstinator) [inline package.json]: might as well pin npm@8 here too

**Review by vanstinator — APPROVED** [review]
ship it

File: tsconfig.json (hunk 2 above)
Comment (dudewheresmycode) [reply tsconfig.json]: I reverted this. Was leftover from debugging the compilation issue.

**Pull request merged**